crypto/ec/ec_pmeth.c

```diff
@@ -323,7 +392,10 @@ static int pkey_ec_ctrl(EVP_PKEY_CTX *ctx, int type, int p1, void *p2)
             EVP_MD_type((const EVP_MD *)p2) != NID_sha224 &&
             EVP_MD_type((const EVP_MD *)p2) != NID_sha256 &&
             EVP_MD_type((const EVP_MD *)p2) != NID_sha384 &&
-            EVP_MD_type((const EVP_MD *)p2) != NID_sha512) {
+            EVP_MD_type((const EVP_MD *)p2) != NID_sha512 &&
+            #ifndef OPENSSL_NO_CNSM
+            EVP_MD_type((const EVP_MD *)p2) != NID_sm3) {
+            #endif
```

Comment:
The ending `) {` should be outside the guard. The way it looks now, build will fail when configured `no-cnsm`

Reply:
(by the way, you need to add the string `"cnsm"` to the `@disablables` array in `Configure`)

Reply:
Yes, it's a mistake indeed. Honestly, we don't test configured with no-cnsm  ever. I think i should check all of those macro define.